ci: do not run benchmark measurements in circleci

devversion opened the pull request:
Currently we run all benchmark perf tests in CircleCI. Since we do not
collect any results, we unnecessarily waste CI/RBE resources. Instead,
we should just not run benchmark perf tests in CI, but still run the
functionality e2e tests which ensure that benchmarks are not broken.

We can do this by splitting the perf and e2e tests into separate
files/targets.

devversion committed:
ci: do not run benchmark measurements in circleci

Currently we run all benchmark perf tests in CircleCI. Since we do not
collect any results, we unnecessarily waste CI/RBE resources. Instead,
we should just not run benchmark perf tests in CI, but still run the
functionality e2e tests which ensure that benchmarks are not broken.

We can do this by splitting the perf and e2e tests into separate
files/targets.

IgorMinar commented on modules/e2e_util/e2e_util.ts:
```diff
@@ -12,7 +12,7 @@ import {browser} from 'protractor';
 import * as webdriver from 'selenium-webdriver';
 
-let cmdArgs: {'bundles': boolean};
+let cmdArgs: {'bundles': boolean}|undefined;
```
I believe that his is obsolete. In the past you used to be able to run the benchmarks with js code bundled or not. with bazel you can only run them as bundled (because the build is fast enough compared to the pre-bazel build).

I suggest, you remove this option.

devversion replied:
Good idea. Will do! I was unsure if this is still needed somewhere.

IgorMinar commented on modules/benchmarks/src/largeform/BUILD.bazel:
```diff
@@ -3,10 +3,21 @@ package(default_visibility = ["//modules/benchmarks:__subpackages__"])
+ts_library(
+    name = "e2e_tests_lib",
+    testonly = 1,
+    srcs = ["largeform.spec.ts"],
```
can we disambiguate the ".spec.ts" files, by calling them ".e2e-spec.ts" and ".perf-spec.ts"?

devversion replied:
Sounds good to me.

IgorMinar commented on modules/benchmarks/e2e_test.bzl:
```diff
@@ -0,0 +1,17 @@
+        on_prepare = "//modules/benchmarks:start-server.js",
+        server = server,
+        # `yargs` is needed as runtime dependency for the e2e utils.
+        deps = ["@npm//yargs"] + deps,
```
do we still need it after you remove the command line option (see comment below)?

devversion replied:
Yeah, this can be removed then. Nice idea 👍

IgorMinar reviewed (changes requested):
thanks @devversion!  I found a few things that I think we could improve, please take a look at the individual comments.

devversion committed:
test: disambiguate e2e and perf spec files

Currently, based on the file names it's not quite clear whether
a given `.spec.ts` file runs benchmark perf or benchmark e2e
functionality tests. To disambiguate these, we use new file
suffixs. i.e. `e2e-spec.ts` and `perf-spec.ts`.

devversion committed:
test: ensure global options for benchmark tests can be set in bazel

Previously, when the benchmark tests ran outside of Bazel, developers
had the posibility to control how the tests run through command line
options. e.g. `--dryrun`. This no longer works reliable in Bazel where
command line arguments are not passed to the text executable.

To make the global options still usable (as they could be still useful
in some cases), we just pass them through the Bazel `--test_env`. This
reduces the code we need to read the command line, but still preserves
the flexibility in a Bazel idiomatic way.

devversion commented:
@IgorMinar Thanks for reviewing! I addressed your feedback.

IgorMinar commented:
thank you!

IgorMinar commented:
Thanks Paul!

IgorMinar commented:
this PR fails in g3 in a catastrophic way with build errors due to internal latency lab tests. we'll need to investigate this and fix it next week.

devversion commented:
I don't have enough insight, but I'd guess it's because we either changed the file names of the spec files, or since we updated the test library target names (depending on how it is set up in g3)

josephperrott commented:
@devversion The change in file names caused the files to be included in a `glob` that they shouldn't have been in.  I created the internal change to prevent them from being included and then we should be able to re-presubmit this and get it merged.

AndrewKushnir commented:
I started a [new g3 presubmit](http://test/OCL:292038789:BASE:292038823:1580257804362:f6241609).

@devversion could you plz rebase your PR on top of the latest master, so we rerun CI with the most up-to-date version?

Thank you.

devversion commented:
@AndrewKushnir Rebased.
